packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -45,12 +52,103 @@ enum SummaryManagerState {
 // If this needs to be changed in future, we should re-evaluate what and how we raise to summarizer
 type StopReason = "parentNotConnected" | "parentShouldNotSummarize" | "disposed";
 type ShouldSummarizeState =
-    | { shouldSummarize: true; shouldStart: boolean; }
+    | { shouldSummarize: true; }
     | { shouldSummarize: false; stopReason: StopReason; };
 
+/**
+ * This class encapsulates logic around tracking the elected summarizer client.
+ * It will handle updated the elected client when a summary ack hasn't been seen
+ * for some configured number of ops.
+ */
+export class ElectedSummarizerManager {
+    private readonly clientElection: IOrderedClientElection;
+    /**
+     * Used to calculate number of ops since last summary ack for the current elected client.
+     * This will be undefined if we do not have enough information or there is no elected summarizer.
+     * When a new client is first elected, this will be initially set to the sequence number of
+     * the first inbound op after they are elected.
+     * When a summary ack comes in, this will be set to the sequence number of the summary ack.
+     */
+    private lastSummaryAckSeqForClient: number | undefined;
+    /** Most recently invalidated elected summarizer client id */
+    private invalidClientId: string | undefined;
+
+    public get electedClientId() {
+        return this.clientElection.electedClient?.clientId;
+    }
+
+    constructor(
+        private readonly logger: ITelemetryLogger,
+        private readonly summaryCollection: IEventProvider<ISummaryCollectionOpEvents>,
+        orderedClientCollection: IOrderedClientCollection,
+        private readonly maxOpsSinceLastSummary: number,
+        private readonly refreshSummarizer: () => void,
+    ) {
+        // On every inbound op, if enough ops pass without seeing a summary ack (per elected client),
+        // elect a new client and log to telemetry.
+        this.summaryCollection.on("default", (op) => {
+            if (this.lastSummaryAckSeqForClient === undefined) {
+                // We don't have enough information to know if a client is not summarizing yet.
+                if (this.electedClientId !== undefined) {
+                    this.lastSummaryAckSeqForClient = op.sequenceNumber;
+                }
```

Comment:
why store the seq of some random op?

Reply:
i think i get it, but it could use more comment describing why this ok, and why it makes the math work, even when each client could have a different initial value here

Reply:
This actually brings up a good point.. in response to this comment, I was trying to tighten it up so that the sequence number that actually triggers the election change would be for-sure set here (i.e. the join op or leave op seq#). But when doing so I realized that we can never get it exactly right without adding the seq# that triggered the most recent election to the summary. So I can get it right for all except the one we initially loaded from: i.e. clients that just connected will have a different value than clients who have been connected since before the election change.

Let me think about this some more.. If we go this route (and care about this being exact), we would need to store a number in the summary per `OrderedClientElection` object- which is just the one used for SummaryManager right now.

Reply:
Okay with serialization, now we keep track of `electionSequenceNumber`, which is just the sequence number at which the currently elected client was elected. This allows us to have a deterministic point in time to work with as our baseline when no summary acks have been observed.